frontend/js/summonerStats.js

```diff
@@ -0,0 +1,218 @@
+}
+
+const getAndDisplayMatches = async() => {
+    const matches = await getMatches()
+    const iconid = await getIcon()
+    displayMatches(matches.data, iconid.data)
+
+}
+
+const getAndDisplayProfile = (info) => {
+    const playerName = document.getElementById("playerName")
+    playerName.innerText = new URL(location.href).searchParams.get("summonerName")
+    const playerIcon = document.getElementById("playerIconimg")
+    playerIcon.setAttribute("src", `https://ddragon.leagueoflegends.com/cdn/12.11.1/img/profileicon/${info[0]}.png`)
+    playerIcon.setAttribute("height", "100px")
+    playerIcon.setAttribute("width", "100px")
+    const playerLevel = document.getElementById("playerLevel")
+    playerLevel.innerText = "level " + info[2]
+}
+
+const getMatchInfo = (match, matchInfo, matchTab, player) => {
+    const gameID = document.createElement("div")
+    const gameCreation = match.info.gameEndTimestamp
+    const date = new Date(gameCreation)
+    const minutes = Math.floor(match.info.gameDuration / 60);
+    const seconds = match.info.gameDuration - minutes * 60;
+    const displayDuration = document.createElement("div")
+    const dateDisplayed = "Game Duration: " + minutes + "m " + seconds + "s"
+    displayDuration.append(dateDisplayed)
+    const humanDateFormat = date.toLocaleString()
+    const finalDate = document.createElement("div")
+    finalDate.append(humanDateFormat)
+    match.info.queueId == "400" && gameID.append("Normal Game")
+    match.info.queueId == "420" && gameID.append("Ranked Solo")
+    match.info.queueId == "440" && gameID.append("Flex 5:5 Rank")
+    const win = document.createElement("div")
+    matchInfo.append(gameID)
+    matchInfo.append(finalDate)
+    matchInfo.append(displayDuration)
+    console.log(player.win)
+    win.append(player.win === true ? "Victory" : "Defeat")
+    console.log(win.innerText)
+    matchTab.append(matchInfo)
+    matchInfo.append(win)
+}
+
+const createPlayerTab = () => {
+    const playerTab = document.createElement("div")
+    playerTab.setAttribute("class", "tabPlayer")
+    return playerTab
+}
+const createPlayerName = (player) => {
+    const playerName = document.createElement("a")
+    playerName.innerText = player.summonerName
+    playerName.setAttribute("href", `summonerStats.html?summonerName=${playerName.innerText}`)
+    return playerName
+}
+
+const createMatchTab = () => {
+    const matchTab = document.createElement("div")
+    matchTab.setAttribute("class", "test")
+    return matchTab
+}
+const createPlayersLeft = () => {
+    const playersLeft = document.createElement("div")
+    playersLeft.setAttribute("class", "leftPlayers")
+    return playersLeft
+}
+const createPlayersRight = () => {
+    const playersRight = document.createElement("div")
+    playersRight.setAttribute("class", "rightPlayers")
+    return playersRight
+}
+const createPlayerScore = () => {
+    const playerScore = document.createElement("div")
```

Comment:
Almost all these functions are useless (also the ones below), you can replace them with a single function passing some params and based on the existence of the parameters you can add things to your DOM object, then return it,+.
ex:
createDOMComponent(componentType, classes, innerText, url)